src/query/functions/temporal/aggregation.go

```diff
@@ -84,78 +84,101 @@ type aggNode struct {
 }
 
 func (a *aggNode) Process(values []float64) float64 {
-	switch a.op.operatorType {
-	case AvgTemporalType:
-		var mean, count float64
-		for _, v := range values {
-			if !math.IsNaN(v) {
-				count++
-				mean += (v - mean) / count
-			}
-		}
-		return mean
-
-	case CountTemporalType:
-		var count float64
-		for _, v := range values {
-			if !math.IsNaN(v) {
-				count++
-			}
+	return aggFuncs[a.op.operatorType](values)
+}
+
+func avgOverTime(values []float64) float64 {
+	var sum, count float64
+	for _, v := range values {
+		if !math.IsNaN(v) {
+			count++
+			sum += v
 		}
-		return count
-
-	case MinTemporalType:
-		min := math.Inf(1)
-		for _, v := range values {
-			if !math.IsNaN(v) {
-				min = math.Min(min, v)
-			}
+	}
+	if count == 0 {
+		return math.NaN()
+	}
+	return sum / count
+}
+
+func countOverTime(values []float64) float64 {
+	var count float64
+	for _, v := range values {
+		if !math.IsNaN(v) {
+			count++
 		}
-		return min
-
-	case MaxTemporalType:
-		max := math.Inf(-1)
-		for _, v := range values {
-			if !math.IsNaN(v) {
-				max = math.Max(max, v)
-			}
+	}
+	return count
+}
+
+func minOverTime(values []float64) float64 {
+	min := math.Inf(1)
+	for _, v := range values {
+		if !math.IsNaN(v) {
+			min = math.Min(min, v)
 		}
-		return max
-
-	case SumTemporalType:
-		var sum float64
-		for _, v := range values {
-			if !math.IsNaN(v) {
-				sum += v
-			}
+	}
+	if min == math.Inf(1) {
```

Comment:
newlines

Also this might be a bit dumb, but if math.Inf somehow appears in the series, you'd get incorrect results